File: js/038_CountandSay.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    var ret = "1";
    for(var i = 1; i < n; ++i){
        var tmp = "";
        for(var j = 0; j < ret.length; ++j){
            var num = 1;
            while( j < ret.length - 1 && ret[j] === ret[j+1]){
                ++num;
                ++j;
            }
            tmp += num.toString() + ret[j];
        }
        ret = tmp;
    }
    return ret;
};

File: cpp/038_CountandSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ret = "1";
        for(int i = 1; i < n; ++i){
            string tmp;
            for(int j = 0; j < ret.length(); ++j){
                int num = 1;
                while(j < ret.length() - 1 && ret[j] == ret[j+1]){
                    num++;
                    j++;
                } 
                tmp += std::to_string(num) + ret[j];
            }
            ret = tmp;
        }
        return ret;
    }
};